test/functional/Java12andUp/src/org/openj9/test/java_lang_invoke/Test_MethodHandle.java

```diff
@@ -0,0 +1,164 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package org.openj9.test.java_lang_invoke;
+
+import org.openj9.test.java_lang_invoke.helpers.Helper;
+import org.openj9.test.java_lang_invoke.helpers.HelperInterface;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.testng.log4testng.Logger;
+
+import java.lang.constant.MethodHandleDesc;
+import java.lang.invoke.MethodType;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.util.Optional;
+
+/**
+ * This test Java.lang.invoke.MethodHandle API added in Java 12 and later version.
+ * - describeConstable: one test for each constant in DirectMethodHandleDesc.Kind to make sure each type is supported
+ *
+ */
+public class Test_MethodHandle {
```

Comment:
MethodHandleInfo may also help with the testing as we can validate that the MHInfo and the ConstantDesc describe the same info.